### DISCUSSION TITLE
Setting model metadata manually after database connection

### DISCUSSION THREAD

**Discussion by ghost** [collerek/ormar#559] [Q&A] [answered]
In the example code, all the `Meta` inner class have a `database` class variable that is set during the class definition.

How to delay the setting of this field until I actually have created a `Database` object (via `database.Database(db_url)`)?

Setting it to `None` and then manually updating it for each model (which is not super nice either) after a database conn has been established doesn't work since if we import the model class, ormar immediately raises an exception

```
fk_string = to.Meta.tablename + "." + to.get_column_alias(to.Meta.pkname)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

**Comment by ghost** [comment, marked as answer]
Answering own question

It seems model classes can be defined without issue containing a `database` object in the meta class that hasn't been connected to yet. So we can connect later. This thread can be closed.

**Comment by NiklasRosenstein** [comment]
I have the same question but I don't know the database URL string in advance. Is there a clean way to create an uninitialized database and setting the URL connection string later?

What I do now is quiet a nasty workaround, something akin to

```py
async def main():
  config = load_config_from_disk()
  # Need to set DATABASE_URL so it can be read in db.py to pass to the database object.
  os.environ["DATABASE_URL"] = config.database_url
  # Now import the database module.
  from . import db
  # ...
```

And in `db.py`

```py
class BaseMeta(ormar.ModelMeta):
    database_url = os.environ["DATABASE_URL"]
    database = databases.Database(database_url)
    metadata = sqlalchemy.MetaData()
```
